one/src/cbor_value.rs

```diff
@@ -0,0 +1,285 @@
+use anyhow::{anyhow, Error, Result};
+use cid::Cid;
+use enum_as_inner::EnumAsInner;
+use minicbor::{data::Tag, data::Type, Decoder};
+use ordered_float::OrderedFloat;
+use std::{collections::BTreeMap, ops::Index};
+use tracing::debug;
+
+/// CborValue
+/// a enum for working with dynamic values from decoding CBOR values.
+#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Debug, EnumAsInner)]
+pub enum CborValue {
```

Comment:
btw I'm pretty sure that there is already a library for working with Cbor objects.  Filecoin and IPC use CBOR extensively in Rust, I'm sure they have a library we can use.  Fine to stick with this for now to get this PR in, but we should look into if there's an existing library we can use for this.

Reply:
Nathaniel is working on rust serde_dag_cbor to see where we can decode directly to nested structs.